### ISSUE TITLE
Clipping the P matrix when generating an RDPG should throw a warning

### ISSUE THREAD

**Issue by alyakin314** [microsoft/graspologic#359]
## Expected Behavior
when generating an rdpg, if the matrix P ends up having to be clipped, there should be an exception thrown, because, similarly to ASE on an unconnected graph, it almost certainly means one of the two things:
1. there is a bug elsewhere (and it's sometimes tricky to catch)
2. a user does not exactly understand the rdpg and inner product space

this should at least be the case when rescale = False, since if it is Ture - the user is aware of the behavoir. 

alternatively, we could have three options:
True - identical to current
False - identical to current
None (default) - identical to current False, but also throws a warning if rescaling has to be done.

## Actual Behavior
works as it should if one reads the docs, but does not throw the warnings.

## Template Code
```GraSPy
latent1 = np.array([100, 100]).reshape(-1, 1)
p_mat1 = p_from_latent(latent1, rescale=False, loops=False)
X_graph = sample_edges(p_mat1, directed=False, loops=False)
```

## Full Traceback
```pytb
Paste the full traceback in case there is an exception

```

## Your Environment
* Python version: 3.6
* GraSPy version: 0.2.0

**Comment by bdpedigo** [comment]
I seem to recall that when playing around with semipar, if I did ASE -> Phat -> sample RDPG, the Phats that you get (as opposed to P, where you are probably right that people should be generating them to be not outside of [0, 1]) almost always needed this clipping. Or at least frequently enough that when we had this discussion, it seemed like we might be annoyed by a ton of warnings if we did this. I also seem to recall that doing the rescaling was usually worse. 

I am happy to talk about a solution more, I just worry that if someone tries to do ASE -> sample RDPG on real data or even some simulated data, they will get a lot of warnings even when they aren't necessarily doing something wrong but it's just due to the variance in the ASE, for example
